src/makeInfinityAdder.js

```diff
@@ -6,7 +6,18 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let num = 0;
+  return function adder(x) {
+    if (x) {
+      num += x;
+      return adder;
+    }
+    if (!x) {
+      let result = num;
+      num = 0;
+      return result;
+    }
```

Comment:
This `if` can be removed. If code reaches this part, it will always be executed.